### PR TITLE
[18 India] Implements: Variable Revenue Cities, Connection Bonus, Commodity Bonus

### PR DESCRIPTION
**Pull request by EdPovi**
Implements modifications to calculate Variable Revenue Cities, Connection Bonus, Commodity Bonus for train routes.
Check that route begins and ends at a City

- [x] Branch is derived from the latest `master`
- n/a Add the `pins` or `archive_alpha_games` label if this change will break existing games
- [x] Code passes linter with `docker compose exec rack rubocop -a`
- [x] Tests pass cleanly with `docker compose exec rack rake`

## Implementation Notes
Used location names for inclusion checks. 

### Screenshots
image

### Any Assumptions / Hacks
TODO: Add view to company card for commodities, currently stored as an Array on Corporation Class.

### CODE REVIEW

File: lib/engine/game/g_18_india/game.rb [hunk 1]
```diff
@@ -594,6 +601,167 @@ def legal_tile_rotation?(_entity, _hex, _tile)
...
+
+        # TODO: Add visual indicaiton on VIEW for Corporation card for Claimed Commodities
+        def claim_connession(commodities, corporation)
+          return if corporation.commodities.include?(commodities.first)
+
+          commodities.each do |commodity|
+            @unclaimed_commodities.delete(commodity)
+            corporation.commodities << commodity
+          end
+          @log << "#{corporation.name} claims the #{commodities} connesssion"
```
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:679]:
```suggestion
          @log << "#{corporation.name} claims the #{commodities} concession"
```

File: lib/engine/game/g_18_india/game.rb [hunk 2]
```diff
@@ -594,6 +601,167 @@ def legal_tile_rotation?(_entity, _hex, _tile)
...
+          revenue += 70 if visited_location_names.include?('NEPAL') && visited_location_names.include?('MUMBAI')
+          revenue
+        end
+
+        def available_commodities(corporation)
+          @unclaimed_commodities + corporation.commodities
+        end
+
+        # TODO: Add visual indicaiton on VIEW for Corporation card for Claimed Commodities
+        def claim_connession(commodities, corporation)
```
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:672]:
```suggestion
        def claim_concession(commodities, corporation)
```

**Review by michaeljb — COMMENTED** [review]
Looks good, just a couple small typos, and a merge conflict from one of the other PRs being merged.

File: lib/engine/game/g_18_india/game.rb [hunk 3]
```diff
@@ -621,6 +628,167 @@ def legal_tile_rotation?(_entity, _hex, _tile)
...
+          return 0 unless commodity_sources.count.positive?
+
+          revenue = 0
+          commodity_sources.each do |source|
+            bonus = 0
+            case source
+            when 'OIL'
+              # OIL => MUMBAI [D23] + 30
+              bonus = 30 if visited_names.include?('MUMBAI')
+              claim_connession(['OIL'], corporation) if bonus.positive?
```
Comment (michaeljb) [inline lib/engine/game/g_18_india/game.rb:722]: Oh whoops, I should have mentioned that this line and the other calls also need the spelling fix for `claim_concession`

Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb]: Fixed the calls to the misnamed methods.

**Pull request merged**